nima/webclient/webclient/src/main/java/io/helidon/nima/webclient/http1/Http1ClientConnection.java

```diff
@@ -244,4 +279,71 @@ private String certsToString(Certificate[] peerCertificates) {
+                connection.socket.connect(remoteAddress, (int) connection.options.connectTimeout().toMillis());
+                sslSocket.startHandshake();
+                connection.helidonSocket = TlsSocket.client(sslSocket, connection.channelId);
+                if (LOGGER.isLoggable(TRACE)) {
+                    connection.debugTls(sslSocket);
+                }
+            }
+        },
+        PROXY_PLAIN {
+            @Override
+            protected void connect(Http1ClientConnection connection, InetSocketAddress remoteAddress) throws IOException {
+                connection.socket.connect(connection.connectionKey.proxy().address(),
+                        (int) connection.options.connectTimeout().toMillis());
+                int responseCode = connection.connectToProxy(remoteAddress);
+                if (responseCode != Status.OK_200.code()) {
+                    throw new IllegalStateException("Proxy sent wrong HTTP response code: " + responseCode);
+                }
+                connection.helidonSocket = PlainSocket.client(connection.socket, connection.channelId);
+            }
+        },
+        PROXY_HTTPS {
+            @Override
+            protected void connect(Http1ClientConnection connection, InetSocketAddress remoteAddress) throws IOException {
+                connection.socket.connect(connection.connectionKey.proxy().address(),
+                        (int) connection.options.connectTimeout().toMillis());
+                int responseCode = connection.connectToProxy(remoteAddress);
+                if (responseCode != Status.OK_200.code()) {
+                    throw new IllegalStateException("Proxy sent wrong HTTP response code: " + responseCode);
+                }
+                SSLSocket sslSocket = connection.connectionKey.tls().createSocket("http/1.1", connection.socket, remoteAddress);
+                connection.socket = sslSocket;
+                sslSocket.startHandshake();
+                connection.helidonSocket = TlsSocket.client(sslSocket, connection.channelId);
+                if (LOGGER.isLoggable(TRACE)) {
+                    connection.debugTls(sslSocket);
+                }
+            }
+        };
+
+        protected abstract void connect(Http1ClientConnection connection, InetSocketAddress remoteAddress) throws IOException;
+
+        static void configure(Http1ClientConnection connection, InetSocketAddress remoteAddress) throws IOException {
+            EstablishConnection connector;
+            ConnectionKey key = connection.connectionKey;
+            Proxy proxy = key.proxy();
+            if (proxy != null  && !proxy.isNoHosts(URI.create("http://" + remoteAddress.getHostName() + ":" + remoteAddress.getPort()))) {
+                connector = key.tls() != null ? PROXY_HTTPS : PROXY_PLAIN;
+            } else {
+                connector = key.tls() != null ? HTTPS : PLAIN;
+            }
+            connector.connect(connection, remoteAddress);
+        }
```

Comment:
Please do this evaluation in the connect method of Http1ClientConnection

Reply:
I like to have the logic of what 'EstablishConnection' instance to return there, because it makes it more usable, but I don't like that `connector.connect(connection, remoteAddress);`

I will push some changes there, let me know if you like it more.